src/api/query.rs

```diff
@@ -36,14 +36,53 @@ impl QueryHelper {
                 self.client.clone().get_account_info(addr, &block_hash).await?.account_amount
             }
             Address::BakingRewardAccount => {
-                self.client.clone().get_reward_status(&block_hash).await?.baking_reward_account
+                match self.client.clone().get_reward_status(&block_hash).await? {
+                    RewardsOverview::V0 {
+                        data,
+                    } => data.baking_reward_account,
+                    RewardsOverview::V1 {
+                        common,
+                        ..
+                    } => common.baking_reward_account,
+                }
             }
             Address::FinalizationRewardAccount => {
-                self.client
-                    .clone()
-                    .get_reward_status(&block_hash)
-                    .await?
-                    .finalization_reward_account
+                match self.client.clone().get_reward_status(&block_hash).await? {
+                    RewardsOverview::V0 {
+                        data,
+                    } => data.finalization_reward_account,
+                    RewardsOverview::V1 {
+                        common,
+                        ..
+                    } => common.finalization_reward_account,
+                }
+            }
+            Address::FoundationAccrueAccount => {
+                match self.client.clone().get_reward_status(&block_hash).await? {
+                    RewardsOverview::V0 {
+                        ..
+                    } => {
+                        return Err(ApiError::InvalidAccountAddress(
+                            ACCOUNT_ACCRUE_FOUNDATION.to_string(),
+                        ))
+                    }
+                    RewardsOverview::V1 {
+                        foundation_transaction_rewards,
+                        ..
+                    } => foundation_transaction_rewards,
+                }
+            }
+            Address::PoolAccrueAccount(baker_id) => {
+                match self.client.clone().get_pool_status(baker_id, &block_hash).await? {
+                    PoolStatus::BakerPool {
+                        all_pool_total_capital,
+                        ..
+                    } => all_pool_total_capital,
+                    PoolStatus::PassiveDelegation {
+                        all_pool_total_capital,
+                        ..
+                    } => all_pool_total_capital,
+                }
```

Comment:
I believe that here you should be using http://developer.concordium.software/concordium-rust-sdk/concordium_rust_sdk/types/struct.CurrentPaydayBakerPoolStatus.html#structfield.transaction_fees_earned instead of total_pool_capital (accessible via http://developer.concordium.software/concordium-rust-sdk/concordium_rust_sdk/types/enum.PoolStatus.html#variant.BakerPool.field.current_payday_status ) This field will be None if the baker is not active in the current payday, in which case you can return 0.

The accrue account, as I understand your definition here, is the "virtual" account that accrues transaction fees during the payday so that all tokens are accounted for at all times.

Reply:
Done.